api/src/test/java/org/openmrs/module/eptsreports/reporting/intergrated/calculation/pvls/RoutineCalculationTest.java

```diff
@@ -0,0 +1,128 @@
+
+public class RoutineCalculationTest extends BasePatientCalculationTest {
+
+  @Override
+  public PatientCalculation getCalculation() {
+    return Context.getRegisteredComponents(RoutineCalculation.class).get(0);
+  }
+
+  @Override
+  public Collection<Integer> getCohort() {
+    return Arrays.asList(new Integer[] {2, 6, 7, 8, 999, 432});
+  }
+
+  @Override
+  public CalculationResultMap getResult() {
+    PatientCalculation calculation = getCalculation();
+    CalculationResultMap map = new CalculationResultMap();
+
+    CalculationContext evaluationContext = getEvaluationContext();
+    // initiated ART on 2008-08-01 by hiv enrolment and received and vl
+    // result on
+    // 2018-12-12
+    map.put(2, new SimpleResult(false, calculation, evaluationContext));
+    // initiated ART on 2018-06-21 by starting ARV plan observation and vl
+    // result on
+    // 2019-04-02
+    map.put(6, new SimpleResult(true, calculation, evaluationContext));
+    // initiated ART on 2019-01-18 by historical start date observation but
+    // not with
+    // any vl result
+    map.put(7, new SimpleResult(false, calculation, evaluationContext));
+    // initiated ART on 2019-01-21 by first phamarcy encounter observation
+    // but not
+    // with any vl result
+    map.put(8, new SimpleResult(false, calculation, evaluationContext));
+    // initiated ART on 2019-01-20 by ARV transfer in observation but last
+    // vl was in
+    // last 3 months on 2018-12-12
+    map.put(999, new SimpleResult(false, calculation, evaluationContext));
+    // not initiated on ART but last vl is existing on 2018-12-12
+    map.put(432, new SimpleResult(false, calculation, evaluationContext));
+
+    return map;
+  }
+
+  @Before
+  public void initialise() {
+    // initially test with routine criteria1 > 6 && >= 9 months between ART
+    // init & VL date
+    params.put("criteria", PatientsOnRoutineEnum.ADULTCHILDREN);
+  }
+
+  @Test
+  public void calculateRoutineCriteria1WhenBreastFeedingShouldMatchBetween3To6Months() {
+    // test with routine criteria1 > 3 && >= 6 months between ART
+    // init & VL date
+    params.put("criteria", PatientsOnRoutineEnum.BREASTFEEDINGPREGNANT);
+    Obs obs = Context.getObsService().getObs(3777002);
+    Assert.assertEquals(testsHelper.getDate("2018-06-21 00:00:00.0"), obs.getObsDatetime());
+    // session gets auto updated without saving
+    obs.setObsDatetime(testsHelper.getDate("2018-10-21 00:00:00.0"));
+    Assert.assertEquals(
+        getResult().toString(),
+        service.evaluate(getCohort(), getCalculation(), params, getEvaluationContext()).toString());
+  }
+
+  /**
+   * TODO investigate.i hope routine criteria 2 is alright, i find it strange; being on ART for
+   * atleast 12 months with 2 VL with at-least 12 months away from each other when lookup is
+   * constrained to only 12 months from context now!
+   */
+  public void calculateRoutineCriteria2() {
```

Comment:
Lets remove these functions for criteria2 & 3 since they will be covered by MEPTS-127